### PR TITLE
Make an on-screen queue and make the currently playing have an animated bar

### PR DESCRIPTION
**Pull request by ysbrandB**
Implements #57

### CODE REVIEW

File: client/src/views/ProtubeScreen.vue [hunk 1]
```diff
@@ -135,4 +204,39 @@ socket.on("new-video-timestamp", async (newStamp) => {
     if ((await player.getPlayerState()) === 2) player.playVideo();
   }
 });
+
+// retrieving the queue and stop skeletonloading
+socket.on("connect", async () => {
+  const data = await new Promise((resolve) => {
```
Comment (Menkveld-24) [inline client/src/views/ProtubeScreen.vue:210]: You emit the queue-update on a connection in ScreenSocket.js (server) so you don’t need to fetch it here as well.

**Review by Menkveld-24 — CHANGES_REQUESTED** [review]

File: client/src/views/ProtubeScreen.vue (hunk 1 above)
Comment (ysbrandB) [reply client/src/views/ProtubeScreen.vue]: Yes, fixed now :)

File: client/src/views/ProtubeScreen.vue [hunk 2]
```diff
@@ -135,4 +203,26 @@ socket.on("new-video-timestamp", async (newStamp) => {
     if ((await player.getPlayerState()) === 2) player.playVideo();
   }
 });
+
+socket.on("queue-update", (newQueue) => {
+  queueDuration.value = newQueue.duration;
+  queue.value = newQueue.queue;
+});
+
+function updateCurrentlyPlaying() {
```
Comment (Menkveld-24) [inline client/src/views/ProtubeScreen.vue:212]: This is for updating the progressbar but there is a simpler way of doing this by creating a ref variable and assigning that a percentage that updates on the "new-video-timestamp" socket

**Review by Menkveld-24 — CHANGES_REQUESTED** [review]

**Review by Menkveld-24 — APPROVED** [review]
LGTM

**Pull request merged**